File: python/Strings/Anagram.py
def anagram(str1,str2):
    lst={}

    if len(str1)!=len(str2):
        return False
    for i in range(0,len(str1)):
        if str1[i] in lst.keys():
            lst[str1[i]]=lst.get(str1[i])+1
        else:
            lst[str1[i]]=1

    for i in range(0, len(str2)):
        if str2[i] in lst and lst.get(str2[i])>0:
            lst[str2[i]] = lst.get(str2[i]) - 1
        else:
            return False

    return True

def main():
    str1='MARY'
    str2='ARMY'

    if anagram(str1, str2)==True:
        print('Strings are Anagrams')
    else:
        print('Strings are NOT Anagrams')

if __name__=='__main__':
    main()

File: java/Strings/src/Anagram.java
// Write a program check if two strings are Anagrams or NOT.

import java.util.HashMap;
import java.util.Arrays;

public class Anagram {
	
	public static boolean checkAnagram1(String str1,String str2)
	{
		HashMap<Character,Integer> map=new HashMap<Character,Integer>();
		
		if(str1.length()!=str1.length())
			return false;
		
		for(int i=0;i<str1.length();i++)
		{
			char key=str1.charAt(i);
			if(map.containsKey(key))
			{
				int val=map.get(key);
				map.put(key, val+1);
			}
			else
			{
				map.put(key,1);
			}
		}
		
		
		for(int j=0;j<str2.length();j++)
		{
			char key=str2.charAt(j);
			if(map.containsKey(key) && map.get(key)>0)
			{
				int val=map.get(key);
				map.put(key, val-1);
			}
			else
				return false;
		}
		return true;
	}
	
	public static boolean checkAnagram2(String str1,String str2)
	{
		if(str1.length()!=str2.length())
			return false;
		char[] c_arr1=str1.toCharArray();
		char[] c_arr2=str2.toCharArray();
		
		Arrays.sort(c_arr1);
		Arrays.sort(c_arr2);
		
		String t_str1=new String(c_arr1);
		String t_str2=new String(c_arr2);
		
		if(t_str1.equals(t_str2))
			return true;
		else
			return false;
	}
	
	public static void main(String[] args)
	{
		String str1="LISTEN";
		String str2="SILENT";
		if(checkAnagram1(str1,str2))
			System.out.println("Strings are Anagrams");
		else
			System.out.println("Strings are NOT Anagrams");
		
		if(checkAnagram2(str1,str2))
			System.out.println("Strings are Anagrams");
		else
			System.out.println("Strings are NOT Anagrams");
	}
}
